### PR TITLE
Add data providing/processing system with basic implementation

### PR DESCRIPTION
**Pull request by indyteo**
### Type of modification

Types:
- Code
  - Inputs: Provider, (Post)Processor
  - Exceptions

### Changes

Adds data providing/processing system with basic implementation.

#### Feature description

The goal of this system is to separate responsibilities amongst simple classes to achieve the "complex" task of importing data from various data sources.

The structure is the following:
- The `Provider` reads data from the source and returns an iterable of intermediate objects
- The `Processor` transforms each intermediate object into a `Model`
- Each `PostProcessor` alters the model (metadata, geometry, ...) to "standardize" it and make data from different sources uniform

To illustrate how it works, the current data import process have been adapted to fit into this system:
- The `WFS1_1_GML3_1_DataProvider` now implements `Provider` and returns `SimpleFeature`s
- The `SampleImplementation#downloadVectorFeature` method is now split in:
  - `GeoToolsVectorProcessor` which implements `Processor` and transforms `SimpleFeature` into `JTSGeometryModel`
  - `MetadataCopyPostProcessor` which implements `PostProcessor` and copies a metadata into another (effectively changing its name)
  - `MetadataParsePostProcessor` which implements `PostProcessor` and parses a metadata into a given type
  - An anonymous `PostProcessor` which sets a default value to a metadata, and needs to be generalized into something like `MetadataDefaultPostProcessor`

The whole import pipeline is managed by the `SampleImplementation#retrieveDataAndFillModelStore` method, which illustrate how the process works, but will need to be moved into a dedicated class later.

This PR also includes new exceptions:
- `MinalacException`: a fatal error that prevent further execution of the program
- `IgnorableException`: an error that occurred on an element that could be ignored
  - `ModelException`: an `IgnorableException` about a model
- `RetryableException`: an error that occurred during a task that could be retried

Ignorables and retryables exceptions are used as hints about how to handle the exception, but simple exception handling policy can decide to treat them all as fatal errors.

#### Technical changes

Checkstyle is currently missing a configuration option about empty switch cases, and a special suppression had been added to handle this until this issue gets resolved:
https://github.com/checkstyle/checkstyle/issues/9540

Update: The issue may gets resolved soon, with the addition of a new option `allowEmptySwitchCase` in the `WhitespaceAround` check, in version 10.18.2 (next version).

#### Showcase

As often, everything should work the same, as no new data source had been implemented yet.

### Reason

Having very small tasks like this allows to write simple code and achieve the complex goal of importing heterogeneous data sources.

The post-processors can be configured independently for each data source, and the ability to reuse them (because they are generic) avoid large code duplication.

### TODOs

Added: (5)
- `WFS1_1_GML3_1_DataProvider`: The provider currently uses an hard-coded call to `URL#openStream`, which limits our ability to write tests and other systems (such as snapshots). We should replace this by a dedicated object, i.e. "HTTP Requester", given in constructor.
- `SampleImplementation`: (4)
  - Two metadata post-processors have been created (copy and parse) to illustrate how it works. Another one (default) should be created by extracting and generalizing the inline anonymous post-processor. This is an opportunity for a simple task to discover the system.
  - Because the metadata resulting from the post-processors is currently unused (it will be in an upcoming PR), an extra anonymous post-processor have been added to fail if the value is incorrect. Other than revealing potential mistake in current pipeline, it is completely useless and needs to be removed once #24 is merged.
  - Currently, any `IgnorableException` occurring in the process is silently ignored. This behavior should be made customizable using an exception handling policy defining the action to do on such error (log, ignore, fail, ...).
  - Similarly, `RetryableException` are not handled properly, as no retry mechanism had been implemented yet. They are treated like `MinalacException`, as fatal errors.

### Self-checks

- [x] The code has unit tests associated
- [x] The code has Javadoc Comments associated
- [x] Complex / Unexpected code is explained / justified with a small comment
- [x] Relevant documentation inside the `/docs` folder has been updated
- ~~All examples in `docs/usage/Examples.md` work the same (or have been adapted if subject to changes in this PR)~~
- [x] Git history is clean (each commit accomplish a single task and describe it accordingly)
- [x] The texts have been proofread (documentation, error messages, logs, comments...)

### CODE REVIEW

File: src/main/java/com/ignfab/minalac/generator/processors/post/MetadataRemapperPostProcessor.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package com.ignfab.minalac.generator.processors.post;
+
+import com.ignfab.minalac.generator.models.Model;
+
+import java.util.Map;
+
+public class MetadataRemapperPostProcessor implements PostProcessor<Model, Model> {
+    private final Map<String, String> remap;
+
+    public MetadataRemapperPostProcessor(Map<String, String> remap) {
```
Comment (pyrollo) [inline src/main/java/com/ignfab/minalac/generator/processors/post/MetadataRemapperPostProcessor.java]: Pour le renommage d'attribut, il nous faut un simple "copy from to". Dans l'écrasante majorité des cas, on n'aura qu'un seul attribut à renommer, et rarement plus de deux (dans ce cas, deux copy feront le job).

Comment (pyrollo) [reply src/main/java/com/ignfab/minalac/generator/processors/post/MetadataRemapperPostProcessor.java]: (je pense qu'en général il faut faire des postproc aussi simple que possible, ça rendra leur utilisation plus facile).

File: src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParserPostProcessor.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import com.ignfab.minalac.generator.exceptions.MinalacException;
+import com.ignfab.minalac.generator.models.Model;
+
+import java.util.Map;
+import java.util.function.Function;
+
+public class MetadataParserPostProcessor implements PostProcessor<Model, Model> {
+    private final Map<String, ParsingSpec<?>> parsingSpecs;
+
+    public MetadataParserPostProcessor(Map<String, ParsingSpec<?>> parsingSpecs) {
```
Comment (pyrollo) [inline src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParserPostProcessor.java]: Pareil ici, il faudrait un postprocesseur qui ne parse qu'un seul attribut à la fois.
Ça permettrait de simplifier les chaines (par exemple, faire copy A>B parse B-int default B-1 puis copy C>D parse D-float, plutôt que copy (A>B, C>D) puis parse(B-int, D-float) et default B-1)

File: src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParserPostProcessor.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                    case IGNORE, REMOVE_METADATA -> {}
+                    case SKIP_MODEL -> throw new IgnorableException("Missing metadata: " + name);
+                    case ERROR -> throw new MinalacException("Missing metadata: " + name);
+                }
+            }
+        }
+        return model;
+    }
+
+    public record ParsingSpec<T>(
```
Comment (pyrollo) [inline src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParserPostProcessor.java]: Avec un postproc simple, tout ça pourrait être en paramètres du constructeur.

File: src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParserPostProcessor.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            String name = parsing.getKey();
+            ParsingSpec<?> spec = parsing.getValue();
+            if (model.hasMetadata(name)) {
+                Object value = model.getMetadata(name);
+                if (spec.type().isAssignableFrom(value.getClass()))
+                    continue;
+                try {
+                    Object parsed = spec.parser().apply(value);
+                    if (parsed == null) {
+                        if (spec.failWhenParserReturnNull())
```
Comment (pyrollo) [inline src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParserPostProcessor.java]: Ça pourrait être remplacé par un postproc "not null"

File: src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParserPostProcessor.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                    model.setMetadata(name, parsed);
+                } catch (Throwable e) {
+                    switch (spec.failurePolicy()) {
+                        case IGNORE -> {}
+                        case REMOVE_METADATA -> model.setMetadata(name, null);
+                        case SKIP_MODEL -> throw new IgnorableException("Failed to parse metadata: " + name, e);
+                        case ERROR -> throw new MinalacException("Failed to parse metadata: " + name, e);
+                    }
+                }
+            } else if (spec.failWhenMissingMetadata()) {
```
Comment (pyrollo) [inline src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParserPostProcessor.java]: Idem, postproc not null nous éviterait ce paramètre (je pense).

**Review by pyrollo — COMMENTED** [review]
Permières remarques sur le dev en cours.
Pour les postproc je pense qu'il faut vraiement rester KISS, faire des postproc très atomiques pour simplifier le code et faciliter l'écriture du paramétrage (surtout que le paramétrage des sources va être écrit par des humains peut être pas très techos).

D'un autre côté, il va y avoir un truc à réfléchir pour le schedule des sources et des renderers (classe utilitaire ? classes task spéciales ?) pour mettre le code des méthodes de SampleImplementation (probablement à voir plus tard).

File: src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParserPostProcessor.java (hunk 4 above)
Comment (indyteo) [reply src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParserPostProcessor.java]: C'est pas tout à fait pareil, là c'est plutôt une manière de dire "si la fonction de parsing a renvoyé null, on refuse cette valeur". Mais ce refus va donc être soumis à la politique d'échec, qui peut être "ignore", dans quel cas la valeur non parsée restera comme attribut (et éventuellement un autre post-processor pourra la lire, une espèce de 2ème tentative pour les cas chelous dans les données).

En soit je sais pas si ça arrivera, mais au moins comme ça c'est possible, et puis c'est pas grand chose d'avoir l'option pour le faire, quitte à ce qu'elle ne soit utilisée que dans un cas très bizarre qu'on ne rencontrera que dans 5 ans 😄

Comment (indyteo) [reply src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParserPostProcessor.java]: Exemple de donnée à la con qu'il pourrait y avoir :
Un attribut string avec des nombres, et certains ont une "range" en mode `min/max`, mais pas tous. On met le parsing en int, et puis pour ceux qui échouent, il y a un autre post-processeur dédiée à cette source chelou qui va s'occuper de couper sur le `/`, puis parser le min et max et faire un random entre les deux valeur

File: src/main/java/com/ignfab/minalac/generator/inputs/Provider.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    Result<T> provide() throws MinalacException, RetryableException;
+
+    /**
+     * A result wraps elements and a close method.
+     * Once the close method has been invoked, the underlying
+     * resources can be freed, and no new element will be available.
+     *
+     * @param <T> The type of wrapped elements
+     */
+    interface Result<T> extends Iterable<T>, Closeable {}
```
Comment (pyrollo) [inline src/main/java/com/ignfab/minalac/generator/inputs/Provider.java:55]: Je me demande s'il ne faudrait pas plutôt un `Iterator<T>` car l'itérable sous-entend qu'on peut en tirer plusieurs itérateurs, ce qui n'est pas le cas dans l'implémentation de `WFS1_1_GML3_1_DataProvider.FeatureResults`.

File: src/main/java/com/ignfab/minalac/generator/processors/GeoToolsVectorProcessor.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+    @Override
+    public Class<SimpleFeature> acceptedType() {
+        return SimpleFeature.class;
+    }
+
+    @Override
+    public Class<JTSGeometryModel> modelType() {
+        return JTSGeometryModel.class;
+    }
```
Comment (pyrollo) [inline src/main/java/com/ignfab/minalac/generator/processors/GeoToolsVectorProcessor.java:37]: J'ai l'impression que `acceptedType` et `modelType` vont toujours être les mêmes. On pourrait les mettre dans l'interface.

File: src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParsePostProcessor.java [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+/**
+ * Post-processor parsing a metadata value in-place.
+ * @param <T> type of parsed value
+ */
+public class MetadataParsePostProcessor<T> implements PostProcessor<Model, Model> {
+    private final String name;
+    private final Class<T> type;
+    private final Function<Object, ? extends T> parser;
+    private final ParsingFailurePolicy failurePolicy;
+    private final boolean failWhenMissingMetadata;
```
Comment (pyrollo) [inline src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParsePostProcessor.java:18]: Ce paramètre peut être remplacé par un autre postproc qui vérifie que la valeur existe.

File: src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParsePostProcessor.java [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+    private final boolean failWhenParserReturnNull;
```
Comment (pyrollo) [inline src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParsePostProcessor.java:19]: Si `null` signifie une erreur, il faudrait la gérer avec failurePolicy, sinon, faudrait mettre `null` (ou virer la metadonnée).
J'opte plutôt pour la première possibilité: le parser ne doit pas renvoyer `null` et `failWhenParserReturnNull` est géré par la failurePolicy.

File: src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParsePostProcessor.java [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+        this.parser = parser;
+        this.failurePolicy = failurePolicy;
+        this.failWhenMissingMetadata = failWhenMissingMetadata;
+        this.failWhenParserReturnNull = failWhenParserReturnNull;
+    }
+
+    @Override
+    public Class<? super Model> acceptedModelType() {
+        return Model.class;
+    }
```
Comment (pyrollo) [inline src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParsePostProcessor.java:49]: Ici aussi j'ai l'impression que ça va toujours être la même chose et que ça peut se mettre au niveau de l'interface.

File: src/test/java/com/ignfab/minalac/generator/processors/post/MetadataParsePostProcessorTest.java [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+    public void testCapabilities() {
+        MetadataParsePostProcessor<?> postProcessor = new MetadataParsePostProcessor<>(
+            "name",
+            Object.class,
+            Function.identity(),
+            MetadataParsePostProcessor.ParsingFailurePolicy.IGNORE,
+            false,
+            false
+        );
+        assertTrue(postProcessor.acceptedModelType().isAssignableFrom(TestingModel.class), "post-processor accepts any model type");
```
Comment (pyrollo) [inline src/test/java/com/ignfab/minalac/generator/processors/post/MetadataParsePostProcessorTest.java:36]:
```suggestion
        assertTrue(postProcessor.acceptedModelType().isAssignableFrom(Model.class), "post-processor accepts any model type");
```
ou
```suggestion
        assertTrue(postProcessor.acceptedModelType().isAssignableFrom(TestingModel.class), "post-processor accepts test model type");
```

File: src/test/java/com/ignfab/minalac/generator/processors/post/MetadataCopyPostProcessorTest.java [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+            "a", 1,
+            "b", 2,
+            "c", 3
+        ));
+    }
+
+    @Test
+    public void testCapabilities() {
+        MetadataCopyPostProcessor postProcessor = new MetadataCopyPostProcessor("a", "d", false, false);
+        assertTrue(postProcessor.acceptedModelType().isAssignableFrom(TestingModel.class), "post-processor accepts any model type");
```
Comment (pyrollo) [inline src/test/java/com/ignfab/minalac/generator/processors/post/MetadataCopyPostProcessorTest.java:26]:
```suggestion
        assertTrue(postProcessor.acceptedModelType().isAssignableFrom(Model.class), "post-processor accepts any model type");
```
ou
```suggestion
        assertTrue(postProcessor.acceptedModelType().isAssignableFrom(TestingModel.class), "post-processor accepts testing model type");
```

**Review by pyrollo — CHANGES_REQUESTED** [review]
Les post-processors doivent pouvoir être encore simplifiés en fixant des comportements.

File: src/main/java/com/ignfab/minalac/generator/inputs/Provider.java (hunk 6 above)
Comment (indyteo) [reply src/main/java/com/ignfab/minalac/generator/inputs/Provider.java:55]: Ce "problème" est présent avec pas mal de nos interfaces / code où on utilise parfois des itérables alors qu'on n'a qu'un seul itérateur et qu'on ne pourrait pas en avoir un autre, mais on l'utilise pour la simplicité d'écriture (le for-each).

On peut changer, mais on devra écrire la boucle à la main avec un while / next.

File: src/main/java/com/ignfab/minalac/generator/processors/GeoToolsVectorProcessor.java (hunk 7 above)
Comment (indyteo) [reply src/main/java/com/ignfab/minalac/generator/processors/GeoToolsVectorProcessor.java:37]: Malheureusement non on ne peut pas, parce que `T.class` n'est pas valide (ça serait égal à `Object.class`, car les types génériques sont effacés à la compilation). C'est purement une raison technique pour laquelle on est obligé de garder ces types ici (après on peut envisager de remplacer l'interface par une classe abstraite avec des champs initialisés dans le constructeur parent, mais au final ça change pas grand chose ça devra être écrit quelque part dans chaque provider/processor/post-processor)

File: src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParsePostProcessor.java (hunk 8 above)
Comment (indyteo) [reply src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParsePostProcessor.java:18]: Ça va vraiment donner des configurations obèses, enfin selon moi je trouve ça vraiment plus dur d'écrire des YAMLs de 3 kilomètres car la moindre petite étape est découpée, mais bon si les gens qui ne s'y connaissent pas trouvent ça plus simple d'écrire encore plus de YAML, 🤷

File: src/test/java/com/ignfab/minalac/generator/processors/post/MetadataParsePostProcessorTest.java (hunk 11 above)
Comment (indyteo) [reply src/test/java/com/ignfab/minalac/generator/processors/post/MetadataParsePostProcessorTest.java:36]: Non justement, on vérifie qu'il accepte n'importe quel type de modèle en lui en donnant un qu'il ne "connait pas" (je sais pas si c'est clair, mais c'est bien ce que je veux faire)

File: src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParsePostProcessor.java (hunk 10 above)
Comment (indyteo) [reply src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParsePostProcessor.java:49]: Même chose malheureusement on ne peut pas :/

File: src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParsePostProcessor.java (hunk 9 above)
Comment (indyteo) [reply src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParsePostProcessor.java:19]: En fait pour l'instant on est un peu limité par le fait que l'implémentation des métadonnées dans le modèle considère `null` comme l'absence de donnée, ce qui n'est pas forcément 100% forcé. Et puis aussi, il y a le fait que le résultat de parsing mène réellement à l'absence de donnée, mais peut aussi mener à une erreur (et qu'on veuille ignorer le modèle dans ce cas).
Exemple :
On importe du CSV, les métadonnées sont définies depuis les champs du CSV, tout est `String`, on parse une métadonnée qui représente un entier optionnel, et du coup la valeur spéciale `<null>` sera interprêtée comme "absent" (par exemple). On veut donc que le parser puisse renvoyer `null`, et aussi une erreur, et on aurait `failWhenParserReturnNull = false` et `failurePolicy = SKIP_MODEL`.
Bon après là ça pourrait être découpé avec encore un autre post-processor, mais c'est lourd, et il y a peut-être des cas bizarres où c'est la fonction de parsing seule qui pourra décider...
En fait j'ai vraiment la sensation qu'en enlevant ce petit paramètre (qui reste très simple), ça va compliquer les choses derrière :/

File: src/main/java/com/ignfab/minalac/generator/processors/GeoToolsVectorProcessor.java (hunk 7 above)
Comment (pyrollo) [reply src/main/java/com/ignfab/minalac/generator/processors/GeoToolsVectorProcessor.java:37]: Ah ouais, c'est con. Je pense toujours en template plutôt que type erasure. Tant pis.

File: src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParsePostProcessor.java (hunk 8 above)
Comment (pyrollo) [reply src/main/java/com/ignfab/minalac/generator/processors/post/MetadataParsePostProcessor.java:18]: Pour le coup j'y vois un autre avantage, c'est qu'on peut avoir besoin d'un tel check en dehors du parsing.

File: src/test/java/com/ignfab/minalac/generator/processors/post/MetadataParsePostProcessorTest.java (hunk 11 above)
Comment (pyrollo) [reply src/test/java/com/ignfab/minalac/generator/processors/post/MetadataParsePostProcessorTest.java:36]: Ah oui ! Fatigue :D Je vois maintenant.

File: src/main/java/com/ignfab/minalac/generator/inputs/Provider.java (hunk 6 above)
Comment (pyrollo) [reply src/main/java/com/ignfab/minalac/generator/inputs/Provider.java:55]: Au final l'un ou l'autre ne change pas grand chose puisque de toutes façons on ne va utiliser ce Result qu'une fois dans le processeur (donc pas de risque de faire des conneries en gardant Iterable, pas de surcharge non plus avec Iterator).

File: src/main/java/com/ignfab/minalac/generator/exceptions/MinalacException.java [hunk 13]
```diff
@@ -0,0 +1,32 @@
+package com.ignfab.minalac.generator.exceptions;
+
+/**
+ * Exception thrown when a fatal error occurs. This means that nothing can be
+ * done to recover from this situation, and the program should stop working.
+ */
+public class MinalacException extends Exception {
```
Comment (pyrollo) [inline src/main/java/com/ignfab/minalac/generator/exceptions/MinalacException.java]: `GenerationFailedException` ?

File: src/main/java/com/ignfab/minalac/generator/processors/GeoToolsVectorProcessor.java [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+
+    @Override
+    public JTSGeometryModel process(SimpleFeature feature) throws ModelException {
```
Comment (pyrollo) [inline src/main/java/com/ignfab/minalac/generator/processors/GeoToolsVectorProcessor.java]: Est-ce qu'on va vraiement avoir besoin de distinguer cette exception aujourd'hui ?
Si ce n'est que dans un hypothétique cas futur, on peut se contenter d'une `IgnorableException`.

File: src/main/java/com/ignfab/minalac/generator/processors/post/MetadataCopyPostProcessor.java [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+ * This operation can be seen as renaming a metadata, with
+ * the side effect of leaving the original one in place.
+ * <p>
+ * The model same model object is returned after post-processing.
+ */
+public class MetadataCopyPostProcessor implements PostProcessor<Model, Model> {
+    private final String from;
+    private final String to;
+    private final boolean abortIfFromIsAbsent;
+    private final boolean doNotOverwriteExistingTo;
```
Comment (pyrollo) [inline src/main/java/com/ignfab/minalac/generator/processors/post/MetadataCopyPostProcessor.java:17]: On peut le laisser comme ça, mais ça me paraitrait plus clair inversé : juste `overwrite` ou un truc comme ça (dans le yaml je pense que ça va être quelquechose du genre).

**Review by pyrollo — CHANGES_REQUESTED** [review]
Juste trouver un meilleur nom pour `MinalacException` (je proposais `GenerationFailedException` mais ya peut être mieux) et peut être virer `ModelExcetption` pour l'instant si elle n'est pas utile.

File: src/main/java/com/ignfab/minalac/generator/processors/post/MetadataCopyPostProcessor.java (hunk 15 above)
Comment (indyteo) [reply src/main/java/com/ignfab/minalac/generator/processors/post/MetadataCopyPostProcessor.java:17]: Le but était d'avoir les valeurs par défaut à `false` et mettre `true` explicitement

File: src/main/java/com/ignfab/minalac/generator/exceptions/MinalacException.java (hunk 13 above)
Comment (indyteo) [reply src/main/java/com/ignfab/minalac/generator/exceptions/MinalacException.java]: En fait à l'origine j'avais fait toutes les exceptions (ignorable, retryable, ...) qui `extends MinalacException`, et au final j'ai changé, d'où le nom générique

File: src/main/java/com/ignfab/minalac/generator/processors/post/MetadataCopyPostProcessor.java (hunk 15 above)
Comment (pyrollo) [reply src/main/java/com/ignfab/minalac/generator/processors/post/MetadataCopyPostProcessor.java:17]: Ça marche aussi (si on considère que par défaut on n'écrase pas). Sinon `keepExistingTo` ?

Comment (pyrollo) [reply src/main/java/com/ignfab/minalac/generator/processors/post/MetadataCopyPostProcessor.java:17]: D'ailleurs c'est un comportement habituel de n'écraser que si on le spécifie.

File: src/main/java/com/ignfab/minalac/generator/processors/GeoToolsVectorProcessor.java (hunk 14 above)
Comment (indyteo) [reply src/main/java/com/ignfab/minalac/generator/processors/GeoToolsVectorProcessor.java]: En fait je crois qu'à l'origine j'avais mis le modèle dedans, dans le but plus tard de pouvoir log automatiquement des infos utiles sur le modèle qui a provoqué l'erreur, mais au final je me suis ravisé parce qu'au moment où l'exception est lancée, le modèle n'existe pas encore puisque le seul endroit où je l'ai utilisé c'était justement la création du modèle 😅

**Review by pyrollo — APPROVED** [review]

**Pull request merged**